### PR TITLE
Subscribe to Santiment newsletter - setting and API

### PR DESCRIPTION
**Pull request by tspenov**
```graphql
    mutation {
      changeNewsletterSubscription(newsletterSubscription: OFF) {
        newsletterSubscription
      }
    }
```
```graphql
    {
      currentUser {
        id,
        settings {
          newsletterSubscription
        }
      }
    }
```

### CODE REVIEW

**Comment by Partysun** [comment]
@tspenov I think you need to change the description of this PR.

File: lib/sanbase/auth/settings.ex [hunk 1]
```diff
@@ -2,19 +2,36 @@ defmodule Sanbase.Auth.Settings do
...
+    |> validate_change(:newsletter_subscription, &validate_subscription_type/2)
+    |> put_change(
+      :newsletter_subscription,
+      params[:newsletter_subscription] |> Atom.to_string() |> String.upcase()
+    )
   end
+
+  defp validate_subscription_type(_, type) when type in @newsletter_subscription_types, do: :ok
+
+  defp validate_subscription_type(_, type),
```
Comment (kalkov) [inline lib/sanbase/auth/settings.ex]: Should use the multi-line syntax

File: lib/sanbase_web/graphql/schema.ex [hunk 2]
```diff
@@ -1235,13 +1235,21 @@ defmodule SanbaseWeb.Graphql.Schema do
...
 
+    @desc "Allow/Dissallow to receive notifications in email/telegram channel"
     field :settings_toggle_channel, :user_settings do
       arg(:signal_notify_telegram, :boolean)
       arg(:signal_notify_email, :boolean)
       middleware(JWTAuth)
       resolve(&UserSettingsResolver.settings_toggle_channel/3)
     end
 
+    @desc "Subscribe/unsibscribe to Santiment newsletter"
```
Comment (kalkov) [inline lib/sanbase_web/graphql/schema.ex]: typo: unsibscribe

File: test/sanbase_web/graphql/user/user_settings_test.exs [hunk 3]
```diff
@@ -98,45 +60,92 @@ defmodule SanbaseWeb.Graphql.UserSettingsTest do
...
 
   test "returns empty settings if user hasn't any", %{conn: conn} do
-    query = """
+    query = current_user_query()
+    result = conn |> execute(query, "currentUser")
+
+    assert result["settings"] == nil
+  end
+
+  test "change newsletter subscription", %{conn: conn, user: user} do
```
Comment (kalkov) [inline test/sanbase_web/graphql/user/user_settings_test.exs]: This test case should actually be divided into multiple ones. In its current form it is hard to read, easy to break and doesn't follow the typical test flow: 

* setup
* execute
* assert
* teardown

Much better would be to have the following tests:
```elixir
describe "newsletter subscription" do
  test "changes subscription to daily"
  test "changes subscription to weekly"
  test "can turns off subscription"
  test "can handle unknown subscription types" 
end
```

File: lib/sanbase/auth/settings.ex [hunk 4]
```diff
@@ -2,11 +2,14 @@ defmodule Sanbase.Auth.Settings do
...
   import Ecto.Changeset
 
+  @newsletter_subscription_types ["DAILY", "WEEKLY", "OFF"]
+
   embedded_schema do
     field(:signal_notify_email, :boolean, default: false)
     field(:signal_notify_telegram, :boolean, default: false)
     field(:telegram_chat_id, :integer)
     field(:has_telegram_connected, :boolean, virtual: true)
+    field(:newsletter_subscription, :string, default: "WEEKLY")
```
Comment (IvanIvanoff) [inline lib/sanbase/auth/settings.ex]: default `OFF`

Comment (tspenov) [reply lib/sanbase/auth/settings.ex]: @Partysun ⬆️ 
I am also for default`OFF`

**Pull request merged**